### ISSUE TITLE
A new CSS style guide proposal

### ISSUE THREAD

**Issue by davidsingal** [Vizzuality/workflow-style-guide#1]
http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/
http://outlinecss.co.uk/get-started.html

**Comment by adammulligan** [comment]
Yep, I'm very much in favour of using the BEM syntax in our CSS in the future! I used something similar (but simpler) to implement the WRW prototype.

I find that the "cascading" part of "cascading style sheets" is my least favourite, and makes maintenance a real pain. Having a single class like ".story--header-title" is so much nicer than trying to figure out what is creating the styles for ".story header h1".

There's also something to be said about performance gains for simplified selectors on complex pages, check out this (3 year old) talk from a Github engineer: https://speakerdeck.com/jonrohan/githubs-css-performance. In fact, even if you're not interested in using BEM, check out that talk as it's a real eye-opener on taking care to write good HTML and CSS.

**Comment by dhakelila** [comment]
I agree, but the danger I see here is that we can create very long names. But I think it is a good convention to be used when needed.

I would keep this to name css modules and would keep states (like .is-hidden) to set temporal states to items that could change on time or after an user action.

**Comment by adammulligan** [comment]
I'm usually not too worried about long names, most text editors have autocomplete :D

I agree with keeping simplified classes for individual states :+1:

**Comment by agnlez** [comment]
Using BEM syntax, we sacrifice the "cascading" feature of CSS, don't we?

I would like to give a try to BEM syntax, but we should be very strict with naming conventions or it will be a chaos!

PD: very interesting talk, @adammulligan thanks for share!
